lta/rest_server.py

```diff
@@ -0,0 +1,92 @@
+"""
+Long Term Archive REST API server.
+
+Run with `python -m lta.rest_server`.
+"""
+
+import asyncio
+import logging
+
+from tornado.ioloop import IOLoop
+
+from rest_tools.client import json_decode
+from rest_tools.server import RestServer, RestHandler, RestHandlerSetup, authenticated, catch_error
+
+from .config import from_environment
+
+
+EXPECTED_CONFIG = {
+    'LTA_REST_URL': 'localhost',
+    'LTA_REST_PORT': 8080,
+}
+
+def lta_auth(**_auth):
+    """
+    Handle RBAC authorization for LTA.
+
+    Like :py:func:`authenticated`, this requires the Authorization header
+    to be filled with a valid token.  Note that calling both decorators
+    is not necessary, as this decorator will perform authentication
+    checking as well.
+
+    Args:
+        roles (list): The roles to match
+
+    Raises:
+        :py:class:`tornado.web.HTTPError`
+    """
+    def make_wrapper(method):
+        @authenticated
+        @catch_error
+        @wraps(method)
+        async def wrapper(self, *args, **kwargs):
+            roles = _auth.get('roles', [])
+
+            authorized = False
+
+            auth_role = self.auth_data.get('long-term-archive',{}).get('role',None)
+            if roles and auth_role in roles:
+                authorized = True
+            else:
+                logging.info('roles: %r    attrs: %r', roles, attrs)
+                logging.info('token_role: %r', auth_role)
+                logging.info('role mismatch')
+
+            if not authorized:
+                raise tornado.web.HTTPError(403, reason="authorization failed")
+
+            return await method(self, *args, **kwargs)
+        return wrapper
+    return make_wrapper
+
+class BaseLTAHandler(RestHandler):
```

Comment:
Missing coverage tests

Reply:
I'm unsure why this, and the following, aren't getting covered. They should be, and the test is returning the right answer.  Maybe a bug in the coverage settings.

Reply:
Might be this:
```
tests/test_rest_server.py::test_server_reachability
  /home/dschultz/Documents/github/lta/env/lib/python3.7/site-packages/_pytest/python.py:165: RuntimeWarning: coroutine 'test_server_reachability' was never awaited
    testfunction(**testargs)
```

Reply:
solved with latest commit